controller/setting_controller.go

```diff
@@ -211,15 +211,27 @@ func (sc *SettingController) handleErr(err error, key interface{}) {
 }
 
 func (sc *SettingController) syncSetting(key string) (err error) {
-	defer func() {
-		err = errors.Wrapf(err, "failed to sync setting for %v", key)
-	}()
-
 	_, name, err := cache.SplitMetaNamespaceKey(key)
 	if err != nil {
 		return err
 	}
 
+	defer func() {
+		err = errors.Wrapf(err, "failed to sync setting for %v", key)
+
+		setting, dsErr := sc.ds.GetSettingExact(types.SettingName(name))
+		if dsErr != nil {
+			sc.logger.WithError(dsErr).Warnf("Failed to get setting: %v", name)
+			return
+		}
+		if setting.Status.Applied != (err == nil) {
```

Comment:
Suggest the following statement for readable code. 

```go
if !setting.Status.Applied && err == nil {
    setting.Status.Applied = true
    // ...
}
```

Reply:
Modified. Thanks.